reports-rest/src/main/java/org/jboss/da/rest/listings/Artifacts.java

```diff
@@ -138,6 +152,11 @@ public ContainsResponse isBlackArtifactPresent(@QueryParam("groupid") String gro
     public SuccessResponse addBlackArtifact(
             @ApiParam(value = "JSON object with keys 'groupId', 'artifactId', and 'version'") RestArtifact artifact) {
         SuccessResponse response = new SuccessResponse();
+        if (whiteService.isArtifactPresent(artifact.getGroupId(), artifact.getArtifactId(),
+                artifact.getVersion())) {
+            whiteService.removeArtifact(artifact.getGroupId(), artifact.getArtifactId(),
```

Comment:
The change looks good! I'm just worried that if between the removal of the artifact from the whitelist, and re-addition to the blacklist, an exception gets thrown and the change to the removal of the artifact from the whitelist is not rolled back (should it be rolled back?)